lib/index.js

```diff
@@ -828,8 +828,13 @@ Stream.prototype.destroy = function () {
     _(this._consumers).each(function (consumer) {
         self._removeConsumer(consumer);
     });
+    _(this._observers).each(function (observer) {
+        self._removeObserver(observer)
+    });
+
     if (this.source) {
         this.source._removeConsumer(this);
+        this.source._removeObserver(this);
```

Comment:
You don't need to call `_removeObserver` here. As of ec7cc9e03548198a0c38e37875f26eb6df4d1a1e, `source` is no longer set for observers.